ts/Series/Venn/VennUtils.ts

```diff
@@ -91,26 +91,31 @@ function addOverlapToSets(
     relations: Array<Highcharts.VennRelationObject>
 ): Array<Highcharts.VennRelationObject> {
     // Calculate the amount of overlap per set.
-    const mapOfIdToProps = relations
+    const mapOfIdToProps: Record<string, Highcharts.VennPropsObject> = {};
+
+    relations
         // Filter out relations consisting of 2 sets.
         .filter((relation): boolean => (relation.sets.length === 2))
         // Sum up the amount of overlap for each set.
-        .reduce(
-            (map, relation): Record<string, Highcharts.VennPropsObject> => {
-                relation.sets.forEach((set, i, arr): void => {
-                    if (!isObject(map[set])) {
-                        map[set] = {
-                            overlapping: {},
-                            totalOverlap: 0
-                        };
+        .forEach((relation): void => {
+            relation.sets.forEach((set, i, arr): void => {
+                if (!isObject(mapOfIdToProps[set])) {
+                    mapOfIdToProps[set] = {
+                        totalOverlap: 0,
+                        overlapping: {}
+                    };
+                }
+
+                mapOfIdToProps[set] = {
+                    totalOverlap: (mapOfIdToProps[set].totalOverlap || 0) +
+                        relation.value,
+                    overlapping: {
+                        ...(mapOfIdToProps[set].overlapping || {}),
+                        [arr[1 - i]]: relation.value
                     }
```

Comment:
I ran into a strange TypeScript issue here. I've tried to check if `typeof mapOfIdToProps[set].totalOverlap !== 'undefined'` and then add `relation.value` to it, but it did not seem to work as the TypeScript always threw 2532 error - `Object is possibly 'undefined'` (no matter how many conditions did I check that it is not `undefined`).